From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Wed, 9 Sep 2015 14:54:38 +0100
Subject: test script exec search

Have the test/run_tests.bash script search for the perceptualdiff
executable more aggressively.  This allows the testing script to work
under a wider variety of build environments, in particular the Debian
debhelper 9 build system which by default creates an obj-XXX directory
for generated files.
---
 test/run_tests.bash | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/test/run_tests.bash b/test/run_tests.bash
index 21d6aae..9f680b1 100755
--- a/test/run_tests.bash
+++ b/test/run_tests.bash
@@ -22,15 +22,20 @@ EOF
 script_directory=$(dirname "$0")
 cd "$script_directory"
 
-if [ -f '../build/perceptualdiff' ]
-then
-	pdiff=../build/perceptualdiff
-elif [ -f '../perceptualdiff' ]
-then
-	pdiff=../perceptualdiff
+found=0
+for d in ../build .. ../obj*; do
+    pdiff="$d/perceptualdiff"
+    if [ -f "$pdiff" ]; then
+	found=1
+	break
+    fi
+done
+
+if [ $found = 0 ]; then
+    echo 'perceptualdiff must be built and exist in the repository root or the "build" directory'
+    exit 1
 else
-	echo 'perceptualdiff must be built and exist in the repository root or the "build" directory'
-	exit 1
+    echo "*** testing executable binary $pdiff"
 fi
 
 #------------------------------------------------------------------------------
